packages/config-array/src/config-array.js

```diff
@@ -949,24 +964,26 @@ export class ConfigArray extends Array {
 		if (!matchFound) {
 			debug(`No matching configs found for ${filePath}`);
 
-			// cache and return result - finalConfig is undefined at this point
-			cache.set(filePath, finalConfig);
-			return finalConfig;
+			// cache and return result
+			configWithStatus = Object.freeze({ status: "unconfigured" });
```

Comment:
Thoughts about preparing these objects in advance, so that we don't create a new instance for each unconfigured (& ignored & external) file?